### PR TITLE
Fix CursorTool issue with LogMapper

### LINKED ISSUE #289: CursorTool does not work with LogMapper for both axes
Using LogMapper for both axes

```
mag = LinePlot(index = ArrayDataSource(w, sort_order='ascending'),
                        value = ArrayDataSource(H),
                        color = curve_color,
                        name = curve_name,
                        index_mapper = LogMapper(range=view_mag.index_range),
                        value_mapper = LogMapper(range=view_mag.value_range))
```

leads to the error below, when adding a cursor with following lines

```
cursor1 = CursorTool(mag,
                        drag_button="left",
                        show_value_line=False,
                        show_marker=True,
                        line_width=2.0,
                        color='black')
mag.overlays.append(cursor1)
```

The problem is, that LogMapper.map_screen returns a 1-element array, while LinearMapper returns a scalar value, if only one data point has to be mapped.

Following subclass of LogMapper works for me as a workaround

```
class LogMapperCorr(LogMapper):
    """
    Corrects chaco.log_mapper.LogMapper for the usage with cursors. Cursors need
    a 1D-array iterable to unpack the x and y values and the LogMapper returns
    a 2D-array. Therefore LogMapper.map_screen has to return a scalar, if the length 
    of the data array is 1.
    """

    def __init__(self, *args, **kwargs):
        LogMapper.__init__(self, *args, **kwargs)

    def map_screen(self, data_array):
        arr = LogMapper.map_screen(self, data_array)

        if len(arr)==1:
            return arr[0]
        else:
            return arr
```

```
Traceback (most recent call last):
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\qt4\base_window.py", line 198, in paintEvent
    self.handler.paintEvent(event)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\qt4\base_window.py", line 50, in paintEvent
    self._enable_window._paint(event)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\abstract_window.py", line 467, in _paint
    self.component.draw(gc, view_bounds=(0, 0, size[0], size[1]))
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\component.py", line 424, in draw
    self._draw(gc, view_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\component.py", line 776, in _draw
    self._dispatch_draw(layer, gc, view_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\container.py", line 272, in _dispatch_draw
    component._dispatch_draw(layer, gc, new_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\container.py", line 272, in _dispatch_draw
    component._dispatch_draw(layer, gc, new_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\component.py", line 796, in _dispatch_draw
    handler(gc, view_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\enable\component.py", line 874, in _draw_overlay
    overlay.overlay(self, gc, view_bounds, mode)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\chaco\tools\line_inspector.py", line 106, in overlay
    self.draw(gc, view_bounds)
  File "C:\Users\speschda\AppData\Local\Enthought\Canopy\User\lib\site-packages\chaco\tools\cursor_tool.py", line 183, in draw
    sx, sy = plot.map_screen(self.current_position)
ValueError: need more than 1 value to unpack
```

### PR DESCRIPTION
**Pull request by aaronayres35**
fixes #289 
This PR pulls the content specific to #289 out from PR #726

### CODE REVIEW

File: chaco/tools/tests/test_cursor_tool.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import unittest
```
Comment (rahulporuri) [inline chaco/tools/tests/test_cursor_tool.py:1]: can you add a copyright header here?

**Review by rahulporuri — APPROVED** [review]
LGTM.

Looks like this change was already made to `CursorTool2D` which is used for plots which inherit from `Base2DPlot` but the changes weren't applied earlier to `CursorTool1D` which is used for plots which inherit from `BaseXYPlot`.

**Comment by aaronayres35** [comment]
Ah, similara to #804, this PR currently just makes a switch from one assumption to another.  ie with the changes here, now the CursorTool will not work with LinearMappers.  I will update the PR

File: chaco/plots/lineplot.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def hittest(self, screen_pt, threshold=7.0, return_distance=False):
             # screen_pt is one of the points in the lineplot
             data_pt = (self.index.get_data()[ndx], self.value.get_data()[ndx])
             if return_distance:
-                scrn_pt = self.map_screen(data_pt)
+                scrn_pt = self.map_screen(data_pt)[0]
```
Comment (aaronayres35) [inline chaco/plots/lineplot.py:125]: Note this change is also made in #802 

Both the changes to `BaseXYPlot1.map_screen` done there and `LinearMapper.map_screen` here can cause the return type to be an an array of shape (N,2).

**Comment by aaronayres35** [comment]
Unfortunately a lot of these changes are some what related.  The changes to `LinearMapper` made here could probably equally have gone in in #802

**Review by rahulporuri — APPROVED** [review]
Still LGTM

**Pull request merged**